src/commands/synthetics/reporters/junit.ts

```diff
@@ -0,0 +1,280 @@
+        url: stepDetail.url,
+        ...this.getStepStats(stepDetail),
+      },
+      browser_error: [],
+      error: [],
+      warning: [],
+    }
+    const steps = [mainStep]
+
+    if (stepDetail.subTestStepDetails) {
+      for (const subStepDetail of stepDetail.subTestStepDetails) {
+        steps.push(...this.getStep(subStepDetail))
+      }
+    }
+
+    if (stepDetail.vitalsMetrics) {
+      mainStep.vitals = stepDetail.vitalsMetrics.map((vital) => ({$: vital}))
+    }
+
+    if (stepDetail.browserErrors?.length) {
+      mainStep.browser_error.push(
+        ...stepDetail.browserErrors.map((error) => ({
+          $: {type: error.type, name: error.name},
+          _: error.description,
+        }))
+      )
+    }
+
+    if (stepDetail.error) {
+      mainStep.error.push({
+        $: {type: 'assertion'},
+        _: stepDetail.error,
+      })
+    }
+
+    if (stepDetail.warnings?.length) {
+      mainStep.warning.push(
+        ...stepDetail.warnings.map((warning) => ({
+          $: {type: warning.type},
+          _: warning.message,
+        }))
+      )
+    }
+
+    return steps
+  }
+
+  private getStepStats(step: Step): Stats {
+    const errors = step.browserErrors ? step.browserErrors.length : 0
+
+    return {
+      allowfailures: step.allowFailure ? 1 : 0,
+      assertions: step.subTestStepDetails ? step.subTestStepDetails.length : 1,
+      errors: errors + (step.error ? 1 : 0),
+      failures: step.error ? 1 : 0,
+      skipped: step.skipped ? 1 : 0,
+      tests: step.subTestStepDetails ? step.subTestStepDetails.length : 1,
+      warnings: step.warnings ? step.warnings.length : 0,
+    }
+  }
+
+  private getSuiteStats(results: PollResult[], stats: Stats | undefined = getDefaultStats()): Stats {
+    for (const result of results) {
+      stats = this.getResultStats(result, stats)
+    }
+
+    return stats
+  }
+
+  private getTestSuite(test: Test, result: PollResult): XMLSuite {
+    return {
+      $: {
+        name: test.name,
+        time: result.result.duration! / 1000,
+        timestamp: result.timestamp,
+        ...this.getResultStats(result),
+      },
+      properties: {
+        property: [
+          {$: {name: 'status', value: test.status}},
+          {$: {name: 'public_id', value: test.public_id}},
+          {$: {name: 'check_id', value: result.check_id}},
+          {$: {name: 'result_id', value: result.resultID}},
+          {$: {name: 'type', value: test.type}},
+          {$: {name: 'message', value: test.message}},
+          {$: {name: 'monitor_id', value: test.monitor_id}},
+          {$: {name: 'tags', value: test.tags.join(',')}},
+          {$: {name: 'locations', value: test.locations.join(',')}},
+          {$: {name: 'start_url', value: result.result.startUrl}},
+          {$: {name: 'device', value: result.result.device.id}},
+          {$: {name: 'width', value: result.result.device.width}},
+          {$: {name: 'height', value: result.result.device.height}},
```

Comment:
These values do not always exist depending on the test type (in that case, it is only for browser, `startUrl` only for browser and api-http, else it is `host` for the others, ...)
I do not know how jUnit handles that and also it may try to access undefined properties as `result.result.device` may be undefined, accessing `device.id` should throw an error.

Reply:
Totally forgot about the API test 😭 
Will adapt to cover both type of tests.